### PR TITLE
Restrict returned work orders for DLO operatives

### PR DESCRIPTION
**Pull request by pixeltrix**
DLO operatives can raise repairs so rather than create a new role we give them both agent and contractor permissions.

### CODE REVIEW

**Comment by pixeltrix** [comment]
Required for https://github.com/LBHackney-IT/repairs-hub-frontend/pull/228

File: RepairsApi/V2/Filtering/WorkOrderFilterProvider.cs [hunk 1]
```diff
@@ -43,14 +43,22 @@ private async Task<List<FilterOption>> GetContractors()
             var liveContractors = await _scheduleOfRatesGateway.GetLiveContractors();
             var filterOptions = new List<FilterOption>(liveContractors.Select(c => new FilterOption { Key = c.ContractorReference, Description = c.ContractorName }));
 
-            if (!_currentUserService.HasAnyGroup(UserGroups.Agent, UserGroups.AuthorisationManager, UserGroups.ContractManager))
+            if (_currentUserService.HasAnyGroup(UserGroups.AuthorisationManager, UserGroups.ContractManager))
             {
+                // Authorisation and contract managers can see all work orders
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Filtering/WorkOrderFilterProvider.cs:48]: This restriction will break the flow of the system as users will be able to raise orders for contractors and not have permission to see them

File: RepairsApi/V2/Filtering/WorkOrderFilterProvider.cs [hunk 2]
```diff
@@ -43,14 +43,22 @@ private async Task<List<FilterOption>> GetContractors()
...
+                return filterOptions;
+            }
+            else if (_currentUserService.HasGroup(UserGroups.Agent) && !_currentUserService.HasGroup(UserGroups.Contractor))
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Filtering/WorkOrderFilterProvider.cs:51]: group membership should always add new permissions never subtract existing one or we open ourselves up to a lot of security edge cases and complexity. 
this seems more like a new user type needs to be added

**Review by kylealexanderhackney — CHANGES_REQUESTED** [review]
I maybe missing something but what do these changes have to do with operatives?

**Comment by pixeltrix** [comment]
Doing this in the front end via filters

**Pull request closed without merging**